pylops_mpi/DistributedArray.py

```diff
@@ -273,7 +358,12 @@ def local_shapes(self):
         -------
         local_shapes : :obj:`list`
         """
-        return self.base_comm.allgather(self.local_shape)
+        if self.base_comm is MPI.COMM_WORLD:
```

Comment:
Above you have `if isinstance(self.base_comm, nccl.NcclCommunicator):`, I think it would be good to always have the same pattern, and I personally prefer:

  if self.base_comm is MPI.COMM_WORLD:
```
if isinstance(self.base_comm, MPI.Comm):
    # MPI
else:
    # NCCL
```
to be checked...